lib/game.rb

```diff
@@ -0,0 +1,126 @@
+require 'pry'
+class Game
+
+  attr_accessor :board, :player_1, :player_2, :winner
+
+  def initialize( p1 = nil, p2 = nil, arena = nil)
+    if arena
+      @board = arena
+    else
+      @board = Board.new
+    end
+
+    if p1
+      @player_1 = p1
+    else
+      @player_1 = Human.new("X")
+    end
+
+    if p2
+      @player_2 = p2
+    else
+      @player_2 = Human.new("O")
+    end
+
+  end
+
+  WIN_COMBINATIONS = [
+    [0, 1, 2],
+    [3, 4, 5],
+    [6, 7, 8],
+    [0, 3, 6],
+    [1, 4, 7],
+    [2, 5, 8],
+    [0, 4, 8],
+    [6, 4, 2]
+  ]
+
+  # current_player assumes that the first player was "X"
+  # it uses Board's #turn_count to calculate whose turn it is.
+  def current_player
+    num_moves = self.board.turn_count
+    num_moves % 2 == 0 ? player_1 : player_2
+  end
+
+  def over?
+    self.board.full?
+  end
+
+  def won?
+    WIN_COMBINATIONS.each do |combination| #iterate over WIN_COMBINATIONS
```

Comment:
Here you can use [`Enumerable#detect`](http://ruby-doc.org/core-2.2.3/Enumerable.html#method-i-detect) to return the winning combination or `nil` if none is found.

```ruby
  def won?
    WIN_COMBINATIONS.find do |combo|
      @board.cells[combo[0]] == @board.cells[combo[1]] &&
      @board.cells[combo[0]] == @board.cells[combo[2]] &&
      @board.taken?(combo[0] + 1)
    end
  end
```